application/templates/vpa.yaml

```
@@ -0,0 +1,24 @@
+{{- if and (.Values.vpa).enabled (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1") -}}
```

This could potentially be misleading: if a user sets `vpa.enabled` to `true`, it could still be that there is no `VerticalPodAutoscaler` resource on their cluster, making it so Helm simply doesn't render the template and the user might end up confused as to what is wrong.

I suggest adding a check instead:

```suggestion
{{- if .Values.vpa.enabled -}}

{{- if not (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1/VerticalPodAutoscaler) }}
  {{- fail "There is no VerticalPodAutoscaler resource definition in the target cluster!" }}
{{- end }}

```

Additionally, a unit test should be added to check that such a validation is actually performed.